htdocs/societe/contact.php

```diff
@@ -138,23 +138,19 @@
 if ($action == 'confirm_delete' && $user->hasRight('societe', 'contact', 'delete')) {
 	$id = GETPOST('id', 'int');
 	if (!empty($id) && $socid > 0) {
-		$db->begin();
-
-		$sql = "DELETE t, et FROM ".MAIN_DB_PREFIX."socpeople AS t";
-		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."socpeople_extrafields AS et ON t.rowid = et.fk_object";
-		$sql .= " WHERE t.fk_soc = ".((int) $socid);
```

Comment:
Using the ferch, delete willbreakbthe control on owner of contact.
Why not just removing the left join on the delete ?

Reply:
@eldy i use the "delete" function of contact class, same code inside the contact card for delete action. Maybe add this verification in "delete" function of contact class no ?